boudicca.base/enricher/src/main/kotlin/base/boudicca/enricher/service/osm/OsmEnricher.kt

```diff
@@ -0,0 +1,218 @@
+package base.boudicca.enricher.service.osm
+
+import base.boudicca.Property
+import base.boudicca.SemanticKeys
+import base.boudicca.enricher.service.Enricher
+import base.boudicca.enricher.service.EnricherOrderConstants
+import base.boudicca.fetcher.Fetcher
+import base.boudicca.fetcher.InMemoryFetcherCache
+import base.boudicca.format.ListFormat
+import base.boudicca.format.UrlUtils
+import base.boudicca.format.UrlUtils.encodeURL
+import base.boudicca.model.structured.Key
+import base.boudicca.model.structured.StructuredEvent
+import base.boudicca.model.structured.StructuredEvent.StructuredEventBuilder
+import base.boudicca.model.structured.Variant
+import base.boudicca.model.structured.VariantConstants.SOURCE_VARIANT_NAME
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.ArrayNode
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import org.slf4j.LoggerFactory
+import org.springframework.core.annotation.Order
+import org.springframework.stereotype.Service
+
+
+private const val NOMINATIM_BASE_URL = "https://nominatim.boudicca.events"
+
+@Service
+@Order(EnricherOrderConstants.OsmEnricherOrder)
+class OsmEnricher(
+    private val fetcher: Fetcher = createDefaultFetcher()
+) : Enricher {
+    val sourceVariant = listOf(Variant(SOURCE_VARIANT_NAME, "OSM"))
+    val logger = LoggerFactory.getLogger(this::class.java)!!
+
+    val mapper = jacksonObjectMapper()
+
+    override fun enrich(event: StructuredEvent): StructuredEvent {
+        val osmId: String? = getOrSearchOsmId(event)
```

Comment:
Do we need to explicitely set the type here? can't it be inferred from getOrSearchOsmId?